tt_eager/tensor/tensor_impl.hpp

```diff
@@ -425,12 +425,27 @@ inline void read_data_from_device_buffer(DeviceBuffer device_buffer, vector<T>&
 
 template <typename T, template <typename> typename BufferType>
 inline void write_data_to_device_buffer(
-    CommandQueue& cq, const BufferType<T>& host_buffer, DeviceBuffer device_buffer) {
+    CommandQueue & cq, const BufferType<T>& host_buffer, DeviceBuffer device_buffer, bool writing_owned_storage) {
```

Comment:
Instead of adding `writing_owned_storage` as a separate argument, please use `std::is_same_v` to check whether `BufferType` is of type `borrowed_buffer::Buffer`. Because then it will be a compile-time check

Reply:
Added in latest commit